### PR TITLE
[TACHYON-1131] Use constant strings for preconditions in streams

### PR DESCRIPTION
**Pull request by calvinjia**
https://tachyon.atlassian.net/browse/TACHYON-1131

About an order of magnitude performance gain for operations done in small chunks (< ~100 bytes) per read/write. We may want to consolidate the string constants in the future.

### CODE REVIEW

File: clients/unshaded/src/main/java/tachyon/client/block/BufferedBlockInStream.java [hunk 1]
```diff
@@ -37,6 +37,13 @@
  * Tachyon Stream interfaces.
  */
 public abstract class BufferedBlockInStream extends BlockInStream {
+  // Error strings for preconditions in order to improve performance
+  private static final String ERR_BUFFER_NULL = "Read buffer cannot be null";
+  private static final String ERR_BUFFER_STATE = "Buffer length (%d), offset(%d), len(%d)";
```
Comment (apc999) [inline clients/unshaded/src/main/java/tachyon/client/block/BufferedBlockInStream.java]: I believe only %s accepted rather than %d

Comment (calvinjia) [inline clients/unshaded/src/main/java/tachyon/client/block/BufferedBlockInStream.java]: Oh you're right, thanks.

**Comment by calvinjia** [comment]
Jenkins, test this please.

**Comment by apc999** [comment]
One more failed test:

Expected: (an instance of java.lang.IllegalArgumentException and exception with message a string containing "Seek position is past EOF: 1, fileSize = 0")
     but: exception with message a string containing "Seek position is past EOF: 1, fileSize = 0" message was "Seek position past end of file: 1"
Stacktrace was: java.lang.IllegalArgumentException: Seek position past end of file: 1
 at com.google.common.base.Preconditions.checkArgument(Preconditions.java:119)
 at tachyon.client.file.FileInStream.seek(FileInStream.java:189)
 at tachyon.client.LocalBlockInStreamIntegrationTest.seekExceptionTest2(LocalBlockInStreamIntegrationTest.java:234)

**Pull request merged**